### DISCUSSION TITLE
How to rebase

### DISCUSSION THREAD

**Discussion by mad8vad** [libgit2/libgit2#5784] [Q&A] [answered]
Hi all

How to perform rebase properly? I want move commits from branch "new" to end of branch "master"
Which OIDs should be lookuped to annotate_commits branch, upstream and onto?

git_rebase_init(&out, repo, **branch**, **upstream**, **onto**, opts);

[https://i.imgur.com/CSxaP8T.png](url)
[https://imgur.com/CSxaP8T](url)

Or do I do something completely wrong?
Thanks!

**Comment by personalizedrefrigerator** [comment, marked as answer]
With `git rebase`, I think this would correspond to `git rebase new master` where, from `man git-rebase`,

```
SYNOPSIS
       git rebase [-i | --interactive] [<options>] [--exec <cmd>]
               [--onto <newbase> | --keep-base] [<upstream> [<branch>]]

[  ...  ]

       If <branch> is specified, git rebase will perform an automatic git
       switch <branch> before doing anything else. Otherwise it remains on the
       current branch.
```

`new` is the `<upstream>` and `master` is the `<branch>`.  I think `onto`, here, would be `NULL`...

---

This libgit2 rebase example might be helpful:

https://github.com/personalizedrefrigerator/libgit2/blob/main/examples/rebase.c

**Reply by personalizedrefrigerator** [reply]
At the time of this writing (just after the v1.1 release), interactive rebasing is not supported:

https://github.com/libgit2/libgit2/blob/e16747660f0d21928d2abaa49ac0657c6664f157/src/rebase.c#L373-L382

**Reply by mad8vad** [reply]
Thanks a lot for example! But expression at line #127 may identify that there is nothing to rebase or rebase was fast-forward

**Reply by personalizedrefrigerator** [reply]
See also https://github.com/libgit2/libgit2/issues/3795
